### ISSUE TITLE
Error building docs on latest nightly

### ISSUE THREAD

**Issue by toothbrush7777777** [rustasync/tide#174]
> ```
> $ rustc --version
> rustc 1.35.0-nightly (474e7a648 2019-04-07)
> $ cratesfyi --version
> cratesfyi 0.6.0 (54fff31 2019-04-12)
> $ cratesfyi ...
>     Updating crates.io index
>  Downloading crates ...
>   Downloaded tide v0.1.0
>        Fresh unicode-xid v0.1.0
>        Fresh version_check v0.1.5
>        Fresh semver-parser v0.7.0
>        Fresh autocfg v0.1.2
>        Fresh rand_core v0.4.0
>        Fresh either v1.5.2
>        Fresh cfg-if v0.1.7
>        Fresh stable_deref_trait v1.1.1
>        Fresh lazy_static v1.3.0
>        Fresh nodrop v0.1.13
>        Fresh cc v1.0.35
>        Fresh smallvec v0.6.9
>        Fresh memoffset v0.2.1
>        Fresh scopeguard v0.3.3
>        Fresh lazycell v1.2.1
>        Fresh fnv v1.0.6
>        Fresh futures v0.1.26
>        Fresh pin-utils v0.1.0-alpha.4
>        Fresh slab v0.4.2
>        Fresh siphasher v0.2.3
>        Fresh itoa v0.4.3
>        Fresh string v0.1.3
>        Fresh indexmap v1.0.2
>        Fresh rustc-demangle v0.1.14
>        Fresh try-lock v0.2.2
>        Fresh remove_dir_all v0.5.1
>        Fresh traitobject v0.1.0
>        Fresh safemem v0.2.0
>        Fresh percent-encoding v1.0.1
>        Fresh quick-error v1.2.2
>        Fresh safemem v0.3.0
>        Fresh take_mut v0.2.2
>        Fresh data-encoding v2.1.2
>        Fresh route-recognizer v0.1.12
>        Fresh semver v0.9.0
>        Fresh rand_core v0.3.1
>        Fresh rand_jitter v0.1.3
>     Checking futures-core-preview v0.3.0-alpha.13
>        Fresh log v0.4.6
>        Fresh owning_ref v0.4.0
>      Running `rustc --edition=2018 --crate-name futures_core /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/lib.rs --color never --crate-type lib --emit=dep-info,metadata -C debuginfo=2 --cfg 'feature="either"' --cfg 'feature="std"' -C metadata=2e7e803a8105ff47 -C extra-filename=-2e7e803a8105ff47 --out-dir /home/cratesfyi/cratesfyi/debug/deps -L dependency=/home/cratesfyi/cratesfyi/debug/deps --extern either=/home/cratesfyi/cratesfyi/debug/deps/libeither-bdc8bce0295e6151.rmeta --cap-lints allow`
>        Fresh crossbeam-utils v0.6.5
>        Fresh tokio-trace-core v0.1.0
>        Fresh thread_local v0.3.6
>        Fresh tokio-sync v0.1.4
>        Fresh unsafe-any v0.4.2
>        Fresh proc-macro2 v0.4.27
>        Fresh libc v0.2.51
>        Fresh rustc_version v0.2.3
>        Fresh rand_xorshift v0.1.1
>        Fresh rand_hc v0.1.0
>        Fresh rand_isaac v0.1.1
>        Fresh byteorder v1.3.1
>        Fresh arrayvec v0.4.10
>        Fresh log v0.3.9
>        Fresh want v0.0.6
>        Fresh lock_api v0.1.5
>        Fresh tokio-executor v0.1.7
>        Fresh crossbeam-queue v0.1.2
>        Fresh proc-macro-nested v0.1.3
>        Fresh num-traits v0.2.6
>        Fresh httparse v1.3.3
>        Fresh memchr v2.2.0
>        Fresh slog v2.4.1
>        Fresh ryu v0.2.7
>        Fresh serde v1.0.90
>        Fresh typemap v0.3.3
>        Fresh quote v0.6.12
>        Fresh unicase v1.4.2
>        Fresh rand_pcg v0.1.2
>        Fresh rand_os v0.1.3
>        Fresh iovec v0.1.2
>        Fresh net2 v0.2.33
>        Fresh num_cpus v1.10.0
>        Fresh dirs v1.0.5
>        Fresh rand v0.4.6
>        Fresh time v0.1.42
>        Fresh isatty v0.1.9
>        Fresh rand_chacha v0.1.1
>        Fresh crossbeam-epoch v0.7.1
>        Fresh mime v0.2.6
>        Fresh tokio-timer v0.2.10
>        Fresh tokio-current-thread v0.1.6
>        Fresh backtrace-sys v0.1.28
>        Fresh num-integer v0.1.39
>        Fresh buf_redux v0.8.1
>        Fresh twoway v0.1.8
>     Checking slog-async v2.3.0
>        Fresh serde_json v1.0.39
>        Fresh syn v0.15.31
>        Fresh phf_shared v0.7.24
>        Fresh bytes v0.4.12
>      Running `rustc --crate-name slog_async /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/slog-async-2.3.0/lib.rs --color never --crate-type lib --emit=dep-info,metadata -C debuginfo=2 --cfg 'feature="default"' -C metadata=89607886e7f2b66f -C extra-filename=-89607886e7f2b66f --out-dir /home/cratesfyi/cratesfyi/debug/deps -L dependency=/home/cratesfyi/cratesfyi/debug/deps --extern slog=/home/cratesfyi/cratesfyi/debug/deps/libslog-00bc00346a787279.rmeta --extern take_mut=/home/cratesfyi/cratesfyi/debug/deps/libtake_mut-f8facb6295102b7a.rmeta --extern thread_local=/home/cratesfyi/cratesfyi/debug/deps/libthread_local-0f2d4a7e970b5dcb.rmeta --cap-lints allow`
>        Fresh mio v0.6.16
>        Fresh futures-cpupool v0.1.8
>        Fresh term v0.5.2
>        Fresh tempdir v0.3.7
>        Fresh cookie v0.11.0
>    Compiling rand v0.6.5
>      Running `rustc --crate-name rand /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.6.5/src/lib.rs --color never --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg 'feature="alloc"' --cfg 'feature="default"' --cfg 'feature="rand_core"' --cfg 'feature="rand_jitter"' --cfg 'feature="rand_os"' --cfg 'feature="std"' -C metadata=e659f24eaeae1c5d -C extra-filename=-e659f24eaeae1c5d --out-dir /home/cratesfyi/cratesfyi/debug/deps -L dependency=/home/cratesfyi/cratesfyi/debug/deps --extern libc=/home/cratesfyi/cratesfyi/debug/deps/liblibc-5f4e4d3a4417e1e5.rlib --extern rand_chacha=/home/cratesfyi/cratesfyi/debug/deps/librand_chacha-2bb0e3321678d42f.rlib --extern rand_core=/home/cratesfyi/cratesfyi/debug/deps/librand_core-778c2bd8f98d8595.rlib --extern rand_hc=/home/cratesfyi/cratesfyi/debug/deps/librand_hc-c2e32f93a5572ff1.rlib --extern rand_isaac=/home/cratesfyi/cratesfyi/debug/deps/librand_isaac-91c9549efcba8c53.rlib --extern rand_jitter=/home/cratesfyi/cratesfyi/debug/deps/librand_jitter-4877ba5d3b7e6042.rlib --extern rand_os=/home/cratesfyi/cratesfyi/debug/deps/librand_os-f3b0eee0fb4c0e76.rlib --extern rand_pcg=/home/cratesfyi/cratesfyi/debug/deps/librand_pcg-55bded92c617086b.rlib --extern rand_xorshift=/home/cratesfyi/cratesfyi/debug/deps/librand_xorshift-474f3f45ef6c6a3a.rlib --cap-lints allow --cfg rustc_1_25 --cfg rustc_1_26 --cfg rustc_1_27`
>        Fresh crossbeam-deque v0.7.1
>        Fresh backtrace v0.3.15
>        Fresh chrono v0.4.6
>    Compiling proc-macro-hack v0.5.4
>      Running `rustc --crate-name proc_macro_hack /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro-hack-0.5.4/src/lib.rs --color never --crate-type proc-macro --emit=dep-info,link -C prefer-dynamic -C debuginfo=2 -C metadata=abd93be71659dd13 -C extra-filename=-abd93be71659dd13 --out-dir /home/cratesfyi/cratesfyi/debug/deps -L dependency=/home/cratesfyi/cratesfyi/debug/deps --extern proc_macro2=/home/cratesfyi/cratesfyi/debug/deps/libproc_macro2-aefa8993a8430d7a.rlib --extern quote=/home/cratesfyi/cratesfyi/debug/deps/libquote-d322ea06fa8652c6.rlib --extern syn=/home/cratesfyi/cratesfyi/debug/deps/libsyn-5940bf3b9209cd9d.rlib --cap-lints allow`
> error[E0053]: method `poll` has an incompatible type for trait
>   --> /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/future/future_obj.rs:64:5
>    |
> 64 |     fn poll(self: Pin<&mut Self>, waker: &Waker) -> Poll<T> {
>    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ types differ in mutability
>    |
>    = note: expected type `fn(std::pin::Pin<&mut future::future_obj::LocalFutureObj<'a, T>>, &mut std::task::Context<'_>) -> std::task::Poll<T>`
>               found type `fn(std::pin::Pin<&mut future::future_obj::LocalFutureObj<'a, T>>, &std::task::Waker) -> std::task::Poll<T>`
> 
> error[E0053]: method `poll` has an incompatible type for trait
>    --> /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/future/future_obj.rs:114:5
>     |
> 114 |     fn poll(self: Pin<&mut Self>, waker: &Waker) -> Poll<T> {
>     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ types differ in mutability
>     |
>     = note: expected type `fn(std::pin::Pin<&mut future::future_obj::FutureObj<'a, T>>, &mut std::task::Context<'_>) -> std::task::Poll<T>`
>                found type `fn(std::pin::Pin<&mut future::future_obj::FutureObj<'a, T>>, &std::task::Waker) -> std::task::Poll<T>`
> 
> error: aborting due to 2 previous errors
> 
> For more information about this error, try `rustc --explain E0053`.
> error: Could not compile `futures-core-preview`.
> 
> Caused by:
>   process didn't exit successfully: `rustc --edition=2018 --crate-name futures_core /home/cratesfyi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/lib.rs --color never --crate-type lib --emit=dep-info,metadata -C debuginfo=2 --cfg 'feature="either"' --cfg 'feature="std"' -C metadata=2e7e803a8105ff47 -C extra-filename=-2e7e803a8105ff47 --out-dir /home/cratesfyi/cratesfyi/debug/deps -L dependency=/home/cratesfyi/cratesfyi/debug/deps --extern either=/home/cratesfyi/cratesfyi/debug/deps/libeither-bdc8bce0295e6151.rmeta --cap-lints allow` (exit code: 1)
> warning: build failed, waiting for other jobs to finish...
> thread 'main' panicked at 'ErrorMessage {
>     msg: "build failed"
> }', src/bin/cratesfyi.rs:143:13
> note: Run with `RUST_BACKTRACE=1` for a backtrace.
> ```
> 
> — https://docs.rs/crate/tide/0.1.0/builds/154400

**Comment by toothbrush7777777** [comment]
See also https://github.com/rustasync/tide/issues/173#issuecomment-483427511.

**Comment by fairingrey** [comment]
Once both https://github.com/rustasync/http-service/pull/14 and https://github.com/rustasync/tide/pull/176 are merged, it should build on the latest nightly. Not only that, but the latter PR also updates Travis so the build succeeds too.

In the meantime, as I mentioned in that comment, one can get tide to work on the latest nightly by updating a few things in `Cargo.toml`

```
[...]
futures-preview = "0.3.0-alpha.14"
[...]
[patch.crates-io]
http-service = { git = 'https://github.com/prasannavl/http-service', branch = 'update-futures' }
```

**Comment by prasannavl** [comment]
This should be fixed now with #176 merged.
